flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -802,6 +802,7 @@ void CodeGenAction::generateLLVMIR() {
   pm.enableVerifier(/*verifyPasses=*/true);
 
   MLIRToLLVMPassPipelineConfig config(level, opts, mathOpts);
+  fir::registerDefaultInlinerPass(config);
```

Comment:
Is the pass always registered and executed (irrespective of the optimization level) with the flang-driver? Is that expected and different from tco?

Reply:
Yes, that was the original behavior to register and execute the inliner pass regardless of optimization level with flang-new, tco, and bbc. The changes preserve this behavior.

Reply:
I was confused by the reference to the O2 optimizer pipelines in both bbc and tco. Thanks for clarifying.